.github/workflows/E2E.yaml

```diff
@@ -73,36 +89,81 @@ jobs:
         id: setup
         run: |
           $errorActionPreference = "STOP"
-          . (Join-Path "." "e2eTests/SetupRepositories.ps1") -github -githubOwner '${{ needs.Check.outputs.githubowner }}' -token '${{ Secrets.OrgPAT }}'
+          . (Join-Path "." "e2eTests/SetupRepositories.ps1") -github -githubOwner '${{ needs.Check.outputs.githubowner }}' -token '${{ Secrets.OrgPAT }}' -bcContainerHelperVersion '${{ github.event.inputs.bcContainerHelperVersion }}'
   
-  CalculateReleases:
+  Analyze:
     runs-on: [ ubuntu-latest ]
     needs: [ Check ]
     outputs:
-      releases: ${{ steps.calculatereleases.outputs.releases }}
+      releases: ${{ steps.Analyze.outputs.releases }}
+      scenarios: ${{ steps.Analyze.outputs.scenarios }}
     steps:
       - uses: actions/checkout@v3
         with:
           ref: ${{ github.event.inputs.ref }}
 
-      - name: CalculateReleases
-        id: calculateReleases
+      - name: Analyze
+        id: Analyze
         run: |
+          $errorActionPreference = "STOP"
           $modulePath = Join-Path "." "e2eTests\e2eTestHelper.psm1" -resolve
           Import-Module $modulePath -DisableNameChecking
-          SetTokenAndRepository -githubOwner '${{ needs.Check.outputs.githubowner }}' -token '${{ Secrets.OrgPAT }}' -repository "microsoft/AL-Go" -github
+          SetTokenAndRepository -github -githubOwner '${{ needs.Check.outputs.githubowner }}' -token '${{ Secrets.OrgPAT }}' -repository "microsoft/AL-Go"
           $releases = @(gh release list --repo microsoft/AL-Go | ForEach-Object { $_.split("`t")[0] })
           Write-Host "Releases: $($releases -join ', ')"
-          if ($releases.Count -eq 1) {
+          if ($releases.Count -le 1) {
             $releasesJSon = "[$($releases | ConvertTo-Json -compress)]"
           }
           else {
             $releasesJSon = $releases | ConvertTo-Json -compress
           }
           Add-Content -Path $env:GITHUB_OUTPUT -Value "releases=$releasesJson"
           Write-Host "releases=$releasesJson"
-          Add-Content -Path $env:GITHUB_OUTPUT -Value "releasesCount=$($releases.Count)"
-          Write-Host "releasesCount=$($releases.Count)"
+          $scenarios = @(Get-ChildItem -path (Join-Path $ENV:GITHUB_WORKSPACE "e2eTests/scenarios/*/runtest.ps1") | ForEach-Object { $_.Directory.Name })
```

Comment:
Does `$_.Directory.Name` work well in PS in ubuntu?

Reply:
Yes, works fine